projects/lexical_shared/lib/lexical/document/path.ex

```diff
@@ -125,6 +127,35 @@ defmodule Lexical.Document.Path do
     end
   end
 
+  @spec language_id_from_path(Lexical.path()) :: String.t()
+  def language_id_from_path(path) do
+    case Path.extname(path) do
+      ".ex" ->
+        "elixir"
+
+      ".exs" ->
+        "elixir"
+
+      ".eex" ->
+        "eex"
+
+      ".heex" ->
+        "phoenix-heex"
+
+      extension ->
+        Logger.warning("can't infer lang ID for #{path}, ext: #{extension}.")
+
+        "unsupported (#{extension})"
```

Comment:
🤔 should we include mappings for all the language ids defined here?
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentItem

If so, maybe this belongs in its own module.

Reply:
I think it's best to only map what's supported by Lexical, and handle any unexpected input generically.